backend/src/v1/controllers/wordController.ts

```diff
@@ -1,31 +1,93 @@
 import { Request, Response } from "express";
+import { ProvideWordRepository, WordRepository } from "../data/repositories/wordRepository";
 
-interface WordController {
-    getWords: (req: Request, res: Response) => void;
-    getWord: (req: Request, res: Response) => void;
-    addWord: (req: Request, res: Response) => void;
-    updateWord: (req: Request, res: Response) => void;
-    deleteWord: (req: Request, res: Response) => void;
+export interface WordController {
+    getWords: (req: Request, res: Response) => Promise<void>;
+    getWord: (req: Request, res: Response) => Promise<void>;
+    addWord: (req: Request, res: Response) => Promise<void>;
+    updateWord: (req: Request, res: Response) => Promise<void>;
+    deleteWord: (req: Request, res: Response) => Promise<void>;
 };
 
-export function WordController(): WordController {
-    return WordControllerImpl;
+export function ProvideWordController(): WordController {
+    return new WordControllerImpl(ProvideWordRepository());
 }
 
-const WordControllerImpl: WordController = {
-    getWords: function (req: Request, res: Response): void {
-        res.send("Get all words");
-    },
-    getWord: function (req: Request, res: Response): void {
-        res.send("Get word by ID");
-    },
-    addWord: function (req: Request, res: Response): void {
-        res.send("Add a new word");
-    },
-    updateWord: function (req: Request, res: Response): void {
-        res.send("Update an existing word");
-    },
-    deleteWord: function (req: Request, res: Response): void {
-        res.send("Delete an existing word");
+class WordControllerImpl implements WordController {
+
+    repository: WordRepository = ProvideWordRepository();
+    constructor(repository: WordRepository) {
+        this.repository = repository;
     }
+
+    getWords = async (req: Request, res: Response) => {
```

Comment:
You can avoid a lot of boilerplate and type annotations by calling `router.get` directly:

```ts
wordRouter.get('/words', (req, res) => {
  // req and res have the correct types, no type annotations needed.
});
```

This could be done via a register function, which would give you a way to pass in a `WordRepository`:

```ts
export function registerWordRoutes(router: express.Router, repository: WordRepository) {
  router.get('/words', (req, res) => {
    // you can access repository here
  });
}
```

Reply:
This would change if I remove the WordRepository in favor of top level functions right? i wouldn't need the register since i would just use the functions directly? My thought with making the WordRepository was for testing but i think i was over complicating what the "layers" of the architecture and what should be tested.